### PR TITLE
Cross product as a sink

### PR DESCRIPTION
**Pull request by lnkuiper**
This PR makes the cross product a sink, materializing the pipeline on the right side, similar to the hash join.

It turned out to be a bit trickier than expected, but @Mytherin helped out a ton.

This should allow for better parallelisation in some plans

### CODE REVIEW

File: src/include/duckdb/execution/operator/join/physical_cross_product.hpp [hunk 1]
```diff
@@ -9,18 +9,23 @@
...
 
 namespace duckdb {
 //! PhysicalCrossProduct represents a cross product between two tables
-class PhysicalCrossProduct : public PhysicalOperator {
+class PhysicalCrossProduct : public PhysicalSink {
 public:
 	PhysicalCrossProduct(vector<LogicalType> types, unique_ptr<PhysicalOperator> left,
 	                     unique_ptr<PhysicalOperator> right);
 
+	std::mutex rhs_lock;
```
Comment (Mytherin) [inline src/include/duckdb/execution/operator/join/physical_cross_product.hpp:21]: The lock should be moved tot he GlobalOperatorState; as a rule the operator itself should only contain the constants necessary to describe what needs to be done (e.g. expressions, join conditions, etc). Any tools necessary for performing the operation (intermediates, locks, etc) should be in the operator states.

**Review by Mytherin — COMMENTED** [review]
Thanks! Looks good; one minor comment then I think it can be merged:

File: src/include/duckdb/execution/operator/join/physical_cross_product.hpp (hunk 1 above)
Comment (lnkuiper) [reply src/include/duckdb/execution/operator/join/physical_cross_product.hpp]: Good point. Done!

**Comment by Mytherin** [comment]
Thanks! Looks good now.

**Pull request merged**